integration-tests/src/query_params.test.js

```diff
@@ -93,6 +93,65 @@ describe("for a GET endpoint with no parameters", () => {
   });
 });
 
+const testOptionalEndpoint: SafeAPI.Endpoint<
+  {
+    x: ?string,
+    y: string
+  },
+  {
+    x: ?string,
+    y: string
+  }
+> = SafeAPI.endpoint()
+  .fragment("foo")
+  .queryParams({
+    x: T.maybeString,
+    y: T.string
+  });
+
+const testOptionalHandler = async input => {
+  return {
+    x: input.x ? input.x.toString() : null,
+    y: input.y
+  };
+};
+
+describe("for a GET endpoint with optional query parameters", () => {
+  let server;
+  beforeEach(() => {
+    server = TestUtils.makeServer({
+      endpoint: testOptionalEndpoint,
+      handler: testOptionalHandler
+    });
+  });
+
+  test("server can parse the input when present", async () => {
+    const resp = await fetch(
+      `http://localhost:${server.address().port}/foo?x=X&y=Y`
+    );
+    const json = await resp.json();
+    expect(json).toEqual({ x: "X", y: "Y" });
+  });
+
+  test("server will parse null inputs", async () => {
+    const resp = await fetch(
+      `http://localhost:${server.address().port}/foo?x=&y=Y`
+    );
+    const json = await resp.json();
+    expect(json).toEqual({ x: null, y: "Y" });
+  });
+
+  test("server will accept absent optional inputs", async () => {
+    const resp = await fetch(
+      `http://localhost:${server.address().port}/foo?y=Y`
```

Comment:
Should we be testing with `safeFetch` instead? I would be curious what the deserialization of a serialized null looks like.

Reply:
I'd prefer to keep these unit tests, but I'll add a `foo?x=&y=Y` which is what we're testing for null serialization.

Reply:
never mind. This test is already written above.

Reply:
:+1: Maybe move the test from `merge_query_params` here? That file is checking to see whether two calls to query params merge things correctly (esp. at the type level).